src/Mvc/test/Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -587,9 +596,12 @@ public async Task FromBodyWithInvalidPropertyData_JsonFormatterAddsModelError()
     }
 
     [Theory]
-    [InlineData(false, false)]
-    [InlineData(true, true)]
-    public async Task FromBodyWithEmptyBody_JsonFormatterAddsModelErrorWhenExpected(
+    [InlineData(typeof(Person5), false, false)]
+    [InlineData(typeof(Person5), true, true)]
+    [InlineData(typeof(Person5WithNullableContext), true, false)]
+    [InlineData(typeof(Person5WithNullableContext), false, false)]
+    public async Task FromBodyWithEmptyBody_AddsModelErrorWhenExpected(
+        Type modelType,
         bool allowEmptyInputInBodyModelBindingSetting,
         bool expectedModelStateIsValid)
```

Comment:
This is a bad test because it's representing test branches as data. I realize you didn't write it, but could we split this in to two tests?

Reply:
I have split the test into `FromBodyWithEmptyBody_ModelStateIsInvalid_HasModelErrors` and `FromBodyWithEmptyBody_ModelStateIsValid_WhenAllowEmptyInput`, is that good?